src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,198 @@
+        private const string IncludeClassArgumentName = "include-class";
+
+        protected Dictionary<string, string> bundleArguments;
+
+        protected AndroidInstrumentationBase()
+        {
+        }
+        protected AndroidInstrumentationBase(IntPtr handle, JniHandleOwnership transfer)
+            : base(handle, transfer)
+        {
+        }
+
+        protected virtual IEnumerable<string> Tests
+        {
+            get
+            {
+                yield return Assembly.GetExecutingAssembly().Location;
+            }
+        }
+
+        public override void OnCreate(Bundle arguments)
+        {
+            base.OnCreate(arguments);
+
+            foreach (var key in arguments.KeySet())
+            {
+                string value = arguments.GetString(key);
+                if (!string.IsNullOrEmpty(value))
+                {
+                    bundleArguments.Add(key, value);
+                }
+            }
+
+            if (bundleArguments.ContainsKey(ResultsFileArgumentName) == false)
+            {
+                bundleArguments.Add(ResultsFileArgumentName, "TestResults.xml");
+            }
+
+            Start();
+        }
+
+        public override async void OnStart()
+        {
+            base.OnStart();
+
+            var bundle = new Bundle();
+
+            var entryPoint = new TestsEntryPoint(bundleArguments);
+            entryPoint.TestsCompleted += (sender, results) =>
+            {
+                var message =
+                    $"Tests run: {results.ExecutedTests} " +
+                    $"Passed: {results.PassedTests} " +
+                    $"Inconclusive: {results.InconclusiveTests} " +
+                    $"Failed: {results.FailedTests} " +
+                    $"Ignored: {results.SkippedTests}";
+                bundle.PutString(SummaryMessageArgumentName, message);
+                
+                bundle.PutLong(ExitCodeArgumentName, results.FailedTests == 0 ? 0 : 1);
+            };
+
+            entryPoint.Tests = Tests;
+
+            await entryPoint.RunAsync();
+
+            if (File.Exists(entryPoint.TestsResultsFinalPath))
+            {
+                bundle.PutString(ResultsPathArgumentName, entryPoint.TestsResultsFinalPath);
+            }
+            
+            if (bundle.GetLong(ExitCodeArgumentName, -1) == -1)
+            {
+                bundle.PutLong(ExitCodeArgumentName, 1);
+            }
+
+            Finish(Result.Ok, bundle);
+        }
+
+        class TestsEntryPoint : AndroidApplicationEntryPoint
+        {
+            readonly string resultsPath;
+            readonly string includeMethod;
+            readonly string excludeCategories;
+            readonly string excludeMethod;
+            readonly string includeClass;
+            readonly string excludeClass;
+
+            public TestsEntryPoint(Dictionary<string, string> arguments)
+            {
+                var root = Application.Context.GetExternalFilesDir(null)?.AbsolutePath;
```

Comment:
? implies that root is nullable, right? Looks like it is not considered later in the file, for example, Path.Combine will throw a NRE parameter name path1.

Reply:
This is a good catch. @greenEkatherine can you please turn on the nullability for this project?

It is done by adding `<nullable>enable</nullable>` in the csproj (please check the other csproj, I'm not sure it's `enable` or `enabled`)